### ISSUE TITLE
Replication with Roberta model and GLU task

### ISSUE THREAD

**Issue by YingYellow** [ihounie/lorta#1]
Hi, author. Thank you for your excellent work. I truly appreciate the impressive compression rate presented in your paper. However, while replicating the results in Table 1, I noticed that the accuracy on the **MRPC task** was **68.38** when the rank was set to 8, whereas the paper reports an accuracy of **89.5**.  I was wondering if I might have overlooked something or made a mistake in my replication. Could you kindly provide any suggestions on potential reasons?

The bash code and detailed results are shown below：

<img width="478" alt="Image" src="https://github.com/user-attachments/assets/801cb37d-cfb2-42a5-ab6a-818f6425aeb4" />

<img width="557" alt="Image" src="https://github.com/user-attachments/assets/da8abfb2-2430-413b-81d5-c5aaec823c55" />

**Comment by ihounie** [comment]
Hi Ying, I truly appreciate your interest in our work and efforts in reproducing our results. In GLUE experiments the hyperparameters change across each setting (task and rank), and were set doing a small grid search following prior work as described in the paper. The bash script you point to just has some default hyperparameters. I need to update the bash scripts to match the hyperparameters reported in Appendix E of the paper for each task and setting.

For the setting you mentioned (MRPC, r=8), the hyperparameters reported in the paper are learning rate 1E-02 and classifier learning rate 1E-04, which differ from the ones you used. You can see the wandb run and logs corresponding to the result reported in the paper [here](https://wandb.ai/hounie/peft-glue/runs/mc27bl9t/overview).

**Comment by YingYellow** [comment]
Hi Ignacio. Thank you for sharing the wandb logs. I have updated the **learning rate** to 1E-02 and the **classifier learning rate** to 1E-04. However, the training process still does not converge, and the training loss remains around 0.6.

Would you be able to share the **configuration file** configs/sweep/roberta-large/mrpc/r=8/lr_1E-02_lrc_1E-04.json? That would be extremely helpful. I really appreciate your time and assistance!

<img width="570" alt="Image" src="https://github.com/user-attachments/assets/2e5fc4e1-4aef-4969-b82e-0382949f09c8" />

<img width="1232" alt="Image" src="https://github.com/user-attachments/assets/5997c623-22fd-4c8f-9960-61d88aaebab8" />

<img width="846" alt="Image" src="https://github.com/user-attachments/assets/ab2104b8-a093-4e65-a0a4-a96dd8aed78a" />

**Comment by ihounie** [comment]
My bad, there is a difference in the batch sizes (I used 128 and the bash script has 32). 
This is the config file configs/sweep/roberta-large/mrpc/r=8/lr_1E-02_lrc_1E-04.json :
```
{
    "do_train": true,
    "do_eval": true,
    "gradient_accumulation_steps": 1,
    "output_dir": "./output/model",
    "overwrite_output_dir": true,
    "logging_steps": 10,
    "logging_dir": "./output/log",
    "evaluation_strategy": "epoch",
    "save_strategy": "epoch",
    "warmup_ratio": 0.06,
    "max_grad_norm": 1000.0,
    "weight_decay": 0.0,
    "shared_uv": 1,
    "shared_dim": 1024,
    "model_name_or_path": "roberta-large",
    "per_device_train_batch_size": 128,
    "max_seq_length": 128,
    "mode": "lora",
    "lora_r": 8,
    "init_type": 1,
    "d_init_type": 94,
    "seed": 42,
    "task_name": "mprc",
    "num_train_epochs": 40,
    "classifier_lr": 0.0001,
    "learning_rate": 0.01
} 
```
I need to add the json config files to the repo for all experiments.
